@codex please resolve: 2025-10-05T14:59:39.154Z	Initializing build environment...
2025-10-05T14:59:49.908Z	Success: Finished initializing build environment
2025-10-05T14:59:50.049Z	Cloning repository...
2025-10-05T14:59:51.572Z	Restoring from dependencies cache
2025-10-05T14:59:51.577Z	Restoring from build output cache
2025-10-05T14:59:51.583Z	Detected the following tools from environment: npm@10.9.2, nodejs@22.16.0
2025-10-05T14:59:52.985Z	Success: Dependencies restored from build cache.
2025-10-05T14:59:52.986Z	Installing project dependencies: npm clean-install --progress=false
2025-10-05T15:00:03.462Z	
2025-10-05T15:00:03.463Z	added 183 packages, and audited 184 packages in 10s
2025-10-05T15:00:03.463Z	
2025-10-05T15:00:03.463Z	33 packages are looking for funding
2025-10-05T15:00:03.463Z	  run `npm fund` for details
2025-10-05T15:00:03.487Z	
2025-10-05T15:00:03.487Z	4 moderate severity vulnerabilities
2025-10-05T15:00:03.487Z	
2025-10-05T15:00:03.487Z	To address all issues (including breaking changes), run:
2025-10-05T15:00:03.487Z	  npm audit fix --force
2025-10-05T15:00:03.487Z	
2025-10-05T15:00:03.487Z	Run `npm audit` for details.
2025-10-05T15:00:03.620Z	Executing user build command: npm run build
2025-10-05T15:00:04.405Z	
2025-10-05T15:00:04.405Z	> playwright-mcp-example@0.0.1-next build
2025-10-05T15:00:04.405Z	> tsc
2025-10-05T15:00:04.405Z	
2025-10-05T15:00:06.774Z	Success: Build command completed
2025-10-05T15:00:06.775Z	Executing user deploy command: npx wrangler versions upload
2025-10-05T15:00:08.559Z	
2025-10-05T15:00:08.559Z	 ⛅️ wrangler 4.26.0 (update available 4.42.0)
2025-10-05T15:00:08.559Z	─────────────────────────────────────────────
2025-10-05T15:00:08.634Z	▲ [WARNING] Processing wrangler.toml configuration:
2025-10-05T15:00:08.634Z	
2025-10-05T15:00:08.634Z	    - "unsafe" fields are experimental and may change or break at any time.
2025-10-05T15:00:08.634Z	
2025-10-05T15:00:08.634Z	
2025-10-05T15:00:15.066Z	Total Upload: 5874.41 KiB / gzip: 1117.97 KiB
2025-10-05T15:00:16.317Z	Your Worker has access to the following bindings:
2025-10-05T15:00:16.318Z	Binding                                     Resource             
2025-10-05T15:00:16.318Z	env.MCP_OBJECT (PlaywrightMCP)              Durable Object       
2025-10-05T15:00:16.318Z	env.DB (palywright-mcp-worker)              D1 Database          
2025-10-05T15:00:16.318Z	env.BROWSER                                 Browser              
2025-10-05T15:00:16.318Z	env.assets (ASSETS)                         Unsafe Metadata      
2025-10-05T15:00:16.319Z	
2025-10-05T15:00:16.323Z	
2025-10-05T15:00:16.325Z	✘ [ERROR] A request to the Cloudflare API (/accounts/b3304b14848de15c72c24a14b0cd187d/workers/scripts/playwright-mcp-example/versions) failed.
2025-10-05T15:00:16.326Z	
2025-10-05T15:00:16.326Z	  Cannot use an assets binding without assets. [code: 10305]
2025-10-05T15:00:16.326Z	  
2025-10-05T15:00:16.326Z	  If you think this is a bug, please open an issue at: https://github.com/cloudflare/workers-sdk/issues/new/choose
2025-10-05T15:00:16.326Z	
2025-10-05T15:00:16.326Z	
2025-10-05T15:00:16.440Z	Failed: error occurred while running deploy command
2025-10-05T14:59:39.154Z	Initializing build environment...
2025-10-05T14:59:49.908Z	Success: Finished initializing build environment
2025-10-05T14:59:50.049Z	Cloning repository...
2025-10-05T14:59:51.572Z	Restoring from dependencies cache
2025-10-05T14:59:51.577Z	Restoring from build output cache
2025-10-05T14:59:51.583Z	Detected the following tools from environment: npm@10.9.2, nodejs@22.16.0
2025-10-05T14:59:52.985Z	Success: Dependencies restored from build cache.
2025-10-05T14:59:52.986Z	Installing project dependencies: npm clean-install --progress=false
2025-10-05T15:00:03.462Z	
2025-10-05T15:00:03.463Z	added 183 packages, and audited 184 packages in 10s
2025-10-05T15:00:03.463Z	
2025-10-05T15:00:03.463Z	33 packages are looking for funding
2025-10-05T15:00:03.463Z	  run `npm fund` for details
2025-10-05T15:00:03.487Z	
2025-10-05T15:00:03.487Z	4 moderate severity vulnerabilities
2025-10-05T15:00:03.487Z	
2025-10-05T15:00:03.487Z	To address all issues (including breaking changes), run:
2025-10-05T15:00:03.487Z	  npm audit fix --force
2025-10-05T15:00:03.487Z	
2025-10-05T15:00:03.487Z	Run `npm audit` for details.
2025-10-05T15:00:03.620Z	Executing user build command: npm run build
2025-10-05T15:00:04.405Z	
2025-10-05T15:00:04.405Z	> playwright-mcp-example@0.0.1-next build
2025-10-05T15:00:04.405Z	> tsc
2025-10-05T15:00:04.405Z	
2025-10-05T15:00:06.774Z	Success: Build command completed
2025-10-05T15:00:06.775Z	Executing user deploy command: npx wrangler versions upload
2025-10-05T15:00:08.559Z	
2025-10-05T15:00:08.559Z	 ⛅️ wrangler 4.26.0 (update available 4.42.0)
2025-10-05T15:00:08.559Z	─────────────────────────────────────────────
2025-10-05T15:00:08.634Z	▲ [WARNING] Processing wrangler.toml configuration:
2025-10-05T15:00:08.634Z	
2025-10-05T15:00:08.634Z	    - "unsafe" fields are experimental and may change or break at any time.
2025-10-05T15:00:08.634Z	
2025-10-05T15:00:08.634Z	
2025-10-05T15:00:15.066Z	Total Upload: 5874.41 KiB / gzip: 1117.97 KiB
2025-10-05T15:00:16.317Z	Your Worker has access to the following bindings:
2025-10-05T15:00:16.318Z	Binding                                     Resource             
2025-10-05T15:00:16.318Z	env.MCP_OBJECT (PlaywrightMCP)              Durable Object       
2025-10-05T15:00:16.318Z	env.DB (palywright-mcp-worker)              D1 Database          
2025-10-05T15:00:16.318Z	env.BROWSER                                 Browser              
2025-10-05T15:00:16.318Z	env.assets (ASSETS)                         Unsafe Metadata      
2025-10-05T15:00:16.319Z	
2025-10-05T15:00:16.323Z	
2025-10-05T15:00:16.325Z	✘ [ERROR] A request to the Cloudflare API (/accounts/b3304b14848de15c72c24a14b0cd187d/workers/scripts/playwright-mcp-example/versions) failed.
2025-10-05T15:00:16.326Z	
2025-10-05T15:00:16.326Z	  Cannot use an assets binding without assets. [code: 10305]
2025-10-05T15:00:16.326Z	  
2025-10-05T15:00:16.326Z	  If you think this is a bug, please open an issue at: https://github.com/cloudflare/workers-sdk/issues/new/choose
2025-10-05T15:00:16.326Z	
2025-10-05T15:00:16.326Z	
2025-10-05T15:00:16.440Z	Failed: error occurred while running deploy command

--- a/src/index.ts
+++ b/src/index.ts
@@ -465,18 +465,11 @@ async function serveAsset(env: Env, request: Request, assetPath?: string): Promi
     return new Response('Not Found', { status: 404 });
   }
 
-  const url = new URL(request.url);
-
   if (assetPath) {
+    const url = new URL(request.url);
     url.pathname = assetPath.startsWith('/') ? assetPath : `/${assetPath}`;
+    return env.ASSETS.fetch(new Request(url.toString(), request));
   }
 
-  const assetRequest = assetPath ? new Request(url.toString(), request) : request;
-  const response = await env.ASSETS.fetch(assetRequest);
-
-  if (response.status === 404 && assetPath && !assetPath.endsWith('/index.html')) {
-    return serveAsset(env, request, '/index.html');
-  }
-
-  return response;
+  return env.ASSETS.fetch(request);
 }

